diff --git a/src/Routes.js b/src/Routes.js
index 0b710b0..b46e953 100644
--- a/src/Routes.js
+++ b/src/Routes.js
@@ -31,7 +31,7 @@ const paths = [
 ];
 
 const InsightsRoute = ({ component: Component, rootClass, ...rest }) => {
-    const root = document.getElementById('root');
+    const root = document.getElementById('advisor-root');
     root.removeAttribute('class');
     root.classList.add(`page__${rootClass}`, 'pf-c-page__main');
     root.setAttribute('role', 'main');
diff --git a/src/entry-dev.js b/src/entry-dev.js
index 4e901fc..627b1d2 100644
--- a/src/entry-dev.js
+++ b/src/entry-dev.js
@@ -4,24 +4,33 @@ import { IntlProvider } from '@redhat-cloud-services/frontend-components-transla
 import { NotificationsPortal } from '@redhat-cloud-services/frontend-components-notifications';
 import { Provider } from 'react-redux';
 import React from 'react';
-import ReactDOM from 'react-dom';
 import { BrowserRouter as Router } from 'react-router-dom';
-import { getBaseName } from '@redhat-cloud-services/frontend-components-utilities/files/helpers';
 import { init } from './Store';
 import logger from 'redux-logger';
 import messages from '../locales/data.json';
+import { render, unmountComponentAtNode } from 'react-dom';
+import { initializeApp } from '@scalprum/core';
 
-ReactDOM.render(
-    <IntlProvider locale={navigator.language.slice(0, 2)} messages={messages} onError={console.log}>
-        <Provider store={init(logger).getStore()}>
-            <Router basename={getBaseName(window.location.pathname)}>
-                <React.Fragment>
-                    <NotificationsPortal />
-                    <App />
-                </React.Fragment>
-            </Router>
-        </Provider>
-    </IntlProvider>,
+const url = new URL(document.currentScript.src);
+// eslint-disable-next-line no-undef
+__webpack_public_path__ = `${url.origin}/${__webpack_public_path__}`;
 
-    document.getElementById('root')
-);
+initializeApp({
+    id: 'advisor',
+    name: 'advisor',
+    unmount: () => unmountComponentAtNode(document.getElementById('advisor-root')),
+    update: console.log,
+    mount: () => render(
+        <IntlProvider locale={navigator.language.slice(0, 2)} messages={messages} onError={console.log}>
+            <Provider store={init(logger).getStore()}>
+                <Router basename="/insights/advisor/foo">
+                    <React.Fragment>
+                        <NotificationsPortal />
+                        <App />
+                    </React.Fragment>
+                </Router>
+            </Provider>
+        </IntlProvider>,
+        document.getElementById('advisor-root')
+    )
+});
